[Survey docs][2/n] - Add Creating Surveys page (#6769)
PostHog/posthog.com: contents/docs/surveys/creating-surveys.mdx

---
title: Creating surveys
sidebar: Docs
showTitle: true
availability:
    free: none
    selfServe: full
    enterprise: full
---

To create a new survey, go to the [surveys tab](https://app.posthog.com/surveys) in the PostHog app, and click on the "New survey" button in the top right.

This presents you with a form where you can complete the details of your new survey:

![Create a survey](../../images/docs/surveys/new-survey.png)

Here's a breakdown of each field in the form:

### Display mode

There are two options for displaying a survey:

> 1. **Popover:** Use PostHog's prebuilt user interfaces to show survey's as a popover in the bottom corner of the screen.

Reviewer: ```suggestion
1. **Popover:** Use PostHog's prebuilt user interfaces to show survey's as a popover in the bottom corner of the screen.

```

> 2. **API:** Implement your own survey UI and use PostHog for handling logic and events.

Reviewer: ```suggestion
2. **API:** Implement your own survey UI and use PostHog to handle display logic, submission capture, and analytics.
```

### Question type

> PostHog supports a number of different questions types. All survey types are available for both popover and API display modes.

Reviewer: ```suggestion
PostHog supports multiple questions types. All survey types are available for both popover and API display modes.
```

> Below is a table showing the different questions types:

Reviewer: ```suggestion
```
This is repetitive.

> |                        |                                                                                                       |

Reviewer: Nice table, this is great

|------------------------|-------------------------------------------------------------------------------------------------------|
| **Open text**              | ![Open text survey](../../images/docs/surveys/freetext-survey.png)                                     |
| **Link**                   | ![Link survey](../../images/docs/surveys/link-survey.png)                                              |
| **Rating - emoji**         | ![Emoji rating survey](../../images/docs/surveys/rating-emoji-survey.png)                              |
| **Rating - number**        | ![Number rating survey](../../images/docs/surveys/rating-number-survey.png)                            |
| **Single choice select**   | ![Single choice select survey](../../images/docs/surveys/single-select-survey.png)                      |
| **Multiple choice select** | ![Multiple choice survey](../../images/docs/surveys/multiple-choice-survey.png)                         |

### Targeting

> This specifies the conditions a user must meet in order to be shown the survey. The user must meet **ALL** of the conditions you set to be eligible.

Reviewer: ```suggestion
This specifies the conditions a user must meet to be shown the survey. The user must meet **ALL** the conditions to be eligible.
```

You can target your survey to specific users based on:

- **Linked feature flag:** Whether a user has a specific [feature flag](/docs/feature-flags) enabled. For example, if you're rolling out a new landing page using a feature flag `new-landing-page`, you can gather feedback only from users who have that flag enabled.

> - **Page URL contains:** Where the current URL matches or contains a specific string. For example, you can target a survey to only show on `https://posthog.com/docs`, or show on all docs pages by using the `*` wildcard: `https://posthog.com/docs/*`.

Reviewer 2: So it would just be url always matches as a "contains"

> - **Page URL contains:** Where the current URL matches or contains a specific string. For example, you can target a survey to only show on `https://posthog.com/docs`, or show on all docs pages by using the `*` wildcard: `https://posthog.com/docs/*`.

Author: @liyiy so is it possible to match an exact url e.g. what if I want to match `posthog.com/docs` but not `posthog.com/docs/xyz` ?

> - **Page URL contains:** Where the current URL matches or contains a specific string. For example, you can target a survey to only show on `https://posthog.com/docs`, or show on all docs pages by using the `*` wildcard: `https://posthog.com/docs/*`.

Author: Clarified on slack: Its not possbile

- **Selector matches:** Whether a specific element appears on the page with the specified class name or ID. For example, you can target a survey with `#my-button` or `.my-button` selector.

- **User/group properties:** You can target a survey to users who have a specific [user](/docs/product-analytics/user-properties) or [group property](/docs/product-analytics/group-analytics#how-to-set-group-properties). For example, you can target a survey to users who have a property `is_paying=true`.

![Survey targeting options](../../images/docs/surveys/targeting-options.png)

## Launching your survey

> Once you've set up your survey, click "Save as draft" to save your survey. Then, when you're ready to launch, click the "Launch" button.

Reviewer: Should we mention testing surveys while in draft mode? How do they do it? Otherwise what is the point of draft mode.

Reviewer 2: The point of draft mode is to give users a space to prepare their surveys before launch and to support survey creation because creating a survey that launches immediately with no "save" point would likely cause total abandonment. It's the same as having a toggle button that's "enable/disable this survey" just presented differently. I imagine many users will also want to create and review their surveys before launching them with their team.

Why don't we try saying something like

```suggestion
Once you've set up your survey, click "Save as draft" to save your survey. Here you'll be able to review any changes before launching your survey officially. You may also edit, stop, or resume your survey at any time.
```
